services/wallet/collectibles/manager.go

```diff
@@ -907,7 +907,18 @@ func (o *Manager) checkConnectionStatus(chainID walletCommon.ChainID) {
 			return
 		}
 	}
-	o.statuses[chainID.String()].SetIsConnected(false)
+
+	status, ok := o.statuses[chainID.String()]
+	if !ok {
+		status = connection.NewStatus()
+		state := status.GetState()
+		state.LastSuccessAt = time.Now().Unix()
+		status.SetState(state)
+		status.SetIsConnected(false)
+		o.statuses[chainID.String()] = status
```

Comment:
hmm I'm concerned about thread safety if we're modifying the map here....

Reply:
I agree, we need to avoid potential race conditions here, wondering how it was missed earlier.

Reply:
the map never changed after init originally, and the objects it points to are thread-safe